docs/apigateway.md

```diff
@@ -0,0 +1,110 @@
+
+## Expose an OpenWhisk action
+
+Let's expose a simple action that is already pre-installed with OpenWhisk
+
+```
+$ wsk api-experimental create /hello /echo get /whisk.system/utils/echo
+```
+```
+ok: created api /echo GET for action /whisk.system/utils/echo
+https://21ef035.api-gw.mybluemix.net/hello/echo
+```
+A new URL is generated exposing the `echo` action via a __GET__ HTTP method.
+
+Let's give it a try by sending a HTTP request to the URL.
+```
+$ curl https://21ef035.api-gw.mybluemix.net/hello/echo?marco=polo
+```
+This will invoke the `echo` action, returning back a JSON string with the parameters sent.
+```
+{
+  "marco":"polo"
+}
+```
+
+You can pass parameters to the action via simple query parameters, or via request body.
+
+### Exposing multiple actions
+
+Let's say you want to expose a set of actions for a book club for your friends.
+You have a series of actions to implement your backend for the book club:
+
+| action | http method | description |
+| ----------- | ----------- | ------------ |
+| getBooks    | GET | get book details  |
+| postBooks   | POST | adds a book |
+| putBooks    | PUT | updates book details |
+| deleteBooks | DELETE | deletes a book |
+
+
+Lets create an API for the Book club `Book Club` on basePath `/club` to manage the `books` resources.
+```
+$ wsk api-experimental create -n "Book Club" /club /books get getBooks
+$ wsk api-experimental create /club /books post postBooks
+$ wsk api-experimental create /club /books put putBooks
+$ wsk api-experimental create /club /books delete deleteBooks
+```
+
+Notice that the first action exposed with basepath `/club` gets gets the API label with name `Book Club` any other actions exposed under `/club` will be associated with `Book Club`
+
+Lets list all the actions that we just exposed.
+
+```
+$ wsk api-experimental list
+ok: apis
+Action                   Verb          API Name        URL
+getBooks                   get         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+postBooks                 post         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+putBooks                   put         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+deleteBooks             delete         Book Club       https://2ef15285-gws.api-gw.mybluemix.net/club/books
+```
+
+Now just for fun lets add a new book `JavaScript: The Good Parts` with a HTTP __POST__
+```
+$  curl -X POST -d '{"name":"JavaScript: The Good Parts", "isbn":"978-0596517748"}' https://2ef15285-gws.api-gw.mybluemix.net/club/books
+```
+```
+{
+  "result": "success"
+}
+```
+
+Lets get a list of Books using our action `getBooks` via HTTP __GET__
+```
+$  curl -X GET https://2ef15285-gws.api-gw.mybluemix.net/club/books
+```
+```
+{
+  "result": [{"name":"JavaScript: The Good Parts", "isbn":"978-0596517748"}]
+}
+```
+
+You can delete all the exposed URL using basepath `/club` or API name label `"Book Club"`:
```

Comment:
-> You can delete all of the exposed URLs using either the [base path] `/club` or the API name label `"Book Club"`
.. basepath -> base path